src/adapter/src/coord.rs

```diff
@@ -1509,7 +1509,9 @@ pub async fn serve(
 
             let mut coord = Coordinator {
                 controller: dataflow_client,
-                view_optimizer: Optimizer::logical_optimizer(),
+                view_optimizer: Optimizer::logical_optimizer(
+                    &mz_transform::typecheck::empty_context(),
```

Comment:
Is this still safe for different SQL statements and sessions to reuse?

Reply:
It _should_ be, yes. If someone were to violate the invariant that global identifiers were unique---that is, if we reused global identifiers for different queries---you would get type errors.

To be sure we're getting this right: do the current tests in CI make sure the same `Coordinator` gets used for multiple SQL statements and sessions?

Reply:
It's definitely not a problem then. If for some reason we need to use the same optimizer but reusing global IDs, we could add an API call that would reset optimizer-internal state.

Semi-relatedly, there's been some talk of trying to refactor into a simpler, cleaner optimizer API which might help avoid fiddly context management from optimizer clients.